File: frontend/take_screenshot.js
const puppeteer = require('puppeteer');

(async () => {
  const browser = await puppeteer.launch({ headless: 'new' });
  const page = await browser.newPage();
  
  // Set local storage to mock a logged-in session so we don't get redirected to login
  await page.goto('http://localhost:5173/');
  await page.evaluate(() => {
    localStorage.setItem('auth_token', 'dummy_token'); // Mock auth if needed
    localStorage.setItem('theme', 'dark'); // Force dark mode for cool UI
  });

  await page.setViewport({ width: 1440, height: 900 });
  await page.goto('http://localhost:5173/sentinel', { waitUntil: 'networkidle0' });
  
  // Take screenshot
  await page.screenshot({ path: 'C:/Users/User/.gemini/antigravity/brain/8ca93279-e790-4ae4-b3a8-41b138956926/sentinel_screenshot.png' });
  
  await browser.close();
})();

File: frontend/take_screenshot.py
import asyncio
from playwright.async_api import async_playwright
import sys

async def main():
    screenshot_name = sys.argv[1] if len(sys.argv) > 1 else "verification_1.png"
    async with async_playwright() as p:
        browser = await p.chromium.launch(headless=True)
        context = await browser.new_context(viewport={"width": 1280, "height": 800})
        page = await context.new_page()

        print(f"Navigating to Dashboard to capture {screenshot_name}...")
        await page.goto("http://localhost:5173")
        await page.wait_for_selector(".cc-topbar", timeout=5000)
        
        # Wait a few seconds for the map and metrics to load
        await page.wait_for_timeout(3000)

        out_path = f"C:/Users/User/.gemini/antigravity/brain/15aabb56-dcdf-44fd-be32-72119f59740d/{screenshot_name}"
        await page.screenshot(path=out_path)
        print(f"Captured {screenshot_name}.")

        await browser.close()

if __name__ == "__main__":
    asyncio.run(main())
